### DISCUSSION TITLE
Count Variable

### DISCUSSION THREAD

**Discussion by memoriesbytalha** [community/community#132637] [Programming Help]
### Select Topic Area

General

### Body

import cv2

from ultralytics import YOLO, solutions
import psycopg2
import datetime

# Connect to PostgreSQL database
conn = psycopg2.connect(
    host=db_host,
    port=db_port,
    dbname=db_name,
    user=db_user,
    password=db_password
)
cur = conn.cursor()

# Create table if it doesn't exist
cur.execute("""
    CREATE TABLE IF NOT EXISTS people_count (
        id SERIAL PRIMARY KEY,
        timestamp TIMESTAMP,
        count INTEGER
    )
""")
conn.commit()
model = YOLO("yolov10s.pt")
cap = cv2.VideoCapture("people.mp4")
assert cap.isOpened(), "Error reading video file"
w, h, fps = (int(cap.get(x)) for x in (cv2.CAP_PROP_FRAME_WIDTH, cv2.CAP_PROP_FRAME_HEIGHT, cv2.CAP_PROP_FPS))

line_points = [(20, 400), (1080, 400)]  # line or region points
classes_to_count = [0, 2]  # person and car classes for count

# Video writer
video_writer = cv2.VideoWriter("object_counting_output.avi", cv2.VideoWriter_fourcc(*"mp4v"), fps, (w, h))

# Init Object Counter
class CustomObjectCounter(solutions.ObjectCounter):
    def __init__(self, classes_names):
        super().__init__(
            view_img=False,
            reg_pts=[],
            classes_names=classes_names,
            draw_tracks=True,
            line_thickness=2,
            view_in_counts=True
        )
counter = CustomObjectCounter(model.names)

while cap.isOpened():
    success, im0 = cap.read()
    if not success:
        print("Video frame is empty or video processing has been successfully completed.")
        break
    tracks = model.track(im0, persist=True, show=False, classes=classes_to_count)

    im0 = counter.start_counting(im0, tracks)
    video_writer.write(im0)
    # timestamp = datetime.now()
    # cur.execute("INSERT INTO people_count (timestamp, count) VALUES (%s, %s)", (timestamp))
    # conn.commit()
    
    # video_writer.write(im0)
    import pdb
    pdb.set_trace()
cap.release()
video_writer.release()
cv2.destroyAllWindows()

I am trying to store the total count , but I am unable to identify in which variable the count is stored? anyone who can help?

**Comment by ebndev** [comment]
Hi @memoriesbytalha, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our `Programming Help` category, as this topic might be more relevant there and might help someone give you a nudge in the right direction!

**Comment by ghost** [comment]
Hi, if I understand correctly, the total number is recorded in the "tracks" variable
